### PR TITLE
Update topo_facts from configlet, if present.

### PR DESCRIPTION
**Pull request by renukamanavalan**
### Description of PR
topo_facts.py read info from topology file, which is used by some tests for reference.
e.g. lldp test verifies the neighbor against VMs list in read topo facts.

With configlet, not all VMs are available in topo. So if the topo has a add configlet, 
read the VMs info from configlet too.

Summary:
Fixes # (issue)

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

### CODE REVIEW

File: ansible/library/topo_facts.py [hunk 1]
```diff
@@ -18,19 +19,90 @@
...
 '''
 
+def get_vm_type(ngs_type):
+    if ngs_type == "ToRRouter":
+        return "tor"
+    if ngs_type == "LeafRouter":
+        return "leaf"
+    if ngs_type == "SpineRouter":
+        return "spine"
+
```
Comment (yxieca) [inline ansible/library/topo_facts.py]: Can you add a comment to explain why the translation/conversion is needed?

Why we cannot use the original ngs_type, but have to use a shortened name? I don't see reference in this change so they must have been used by existing code. If there is a good reason for this have to happen, then it should be documented here as comments.

File: ansible/library/topo_facts.py [hunk 2]
```diff
@@ -18,19 +19,90 @@
...
+                    cfg[vm]['intfs'] = []
+                    cfg[vm]['properties'] = ["common"]
+                    cfg[vm]['properties'].append(get_vm_type(e['DEVICE_NEIGHBOR_METADATA'][vm]["type"]))
+
+        # fill data for each VM
+        for e in lst:
+            if e.has_key('DEVICE_NEIGHBOR'):
+                for intf in e['DEVICE_NEIGHBOR']:
+                    vm = e['DEVICE_NEIGHBOR'][intf]['name']
+                    cfg[vm]['interface_indexes'] = [int(intf[-2:]), ]
```
Comment (neethajohn) [inline ansible/library/topo_facts.py]: Vlan index could be incorrect if extracted from interface name. As discussed, please calculate the index from the topo yml

**Comment by neethajohn** [comment]
Thinking about this more, wouldn't it be better to just pass the actual topo yml instead of the clet yml from the script instead of dealing with all this parsing.

File: ansible/library/topo_facts.py (hunk 2 above)
Comment (renukamanavalan) [reply ansible/library/topo_facts.py]: Fixed

File: ansible/library/topo_facts.py (hunk 1 above)
Comment (renukamanavalan) [reply ansible/library/topo_facts.py]: Dropped, by way of re-writing in different fashion.

**Comment by renukamanavalan** [comment]
Re-wrote per Neetha's suggestion. This simplified a lot, with just two simple guidlines
1) Name configlet topo with a suffix "-clet" e.g. "t1-64-lag-clet"
2) Have a corresponding non-configlet version. e.g. "t1-64-lag"

The clet version will have one or more VMs missing compared to corresponding non-clet version. Those missing ones will be added through vars/configlet/<clet topo name>/clet-add.json

**Pull request merged**